Makefile

```diff
@@ -241,3 +245,40 @@ check-manifests: manifests ## Check if 'make manifests' was run.
 .PHONY: ec
 ec: editorconfig ## Run file formatter checks against all project's files.
 	$(EC)
+
+.PHONY: check-images
+check-images: MANAGER_IMAGE:=$(shell grep -I 'newName: ' ./config/manager/kustomization.yaml | awk -F': ' '{print $$2}')
+check-images: MANAGER_TAG:=$(shell grep -I 'newTag: ' ./config/manager/kustomization.yaml | awk -F': ' '{print $$2}')
+check-images: ## Check if images are the same in Makefile and config/manager/kustomization.yaml
+	@if [ "$(MANAGER_IMAGE)" == "" ];then echo "No manager image found" && exit 1;fi
+	@if [ "$(MANAGER_TAG)" == "" ];then echo "No manager tag found" && exit 1;fi
+	@grep -Iq "IMG ?= $(MANAGER_IMAGE):$(MANAGER_TAG)" ./Makefile || (echo "Images differ" && exit 1)
+
+# A valid oadp-operator git repo fork (for example https://github.com/openshift/oadp-operator)
+OADP_FORK ?= openshift
+# A valid branch or tag from oadp-operator git repo
+OADP_VERSION ?= master
+# namespace to deploy development OADP operator
+OADP_NAMESPACE ?= openshift-adp
+
+.PHONY: deploy-dev
+deploy-dev: TEMP:=$(shell mktemp -d)
```

Comment:
Question: Do you need these `deploy-dev` and `undeploy-dev` commands ? 
you would not be using this controller independently, right ? Also, if you wan to run controller locally, shouldn't `make run` suffice ?

Reply:
As specified in command description "Build and push development controller image from current branch and deploy development OADP operator using that image" this does not just create controller deployment in cluster, but an OADP operator with an non-admin image/CRDs with all changes from the current branch of `oadp-non-admin` the command is run from

example: when we create restore controller, we can use this command to test if it works with OADP, without an OADP PR before hand

I would suggest `make build-installer` instead of `make run` for just non-admin controller tests, as documented in architecture https://github.com/migtools/oadp-non-admin/pull/16/files#diff-140eef3ba41bdcf401d507408084181f2c0ac627532b61e0f7906ea7cc926782R7

Reply:
For any integration specific change between OADP Operator and NAC we will definitely need a PR at both the repos and that too at the same time. That means we will be testing OADP Operator and NAC at the same time in these scenarios and OADP Operator already has a `make deploy-olm` command that will install current branch changes (Operator, CRD etc) on your cluster. (You would be using `unsupportedOverrides` for NAC controller image) . Inter-repository changes are not new in our OADP feature development, plugin changes are also of the similar category sometimes. This is the perspective that I am looking from and I believe `make deploy-olm` from OADP Operator should be sufficient. Let me know your thoughts. Do you agree and share similar sentiment ?

Reply:
for me, this command is the most important thing of this PR, allowing us to easy and early test NAC

But I do agree about `make deploy-olm`, we will use it in this command!

Basically this command just do 3 things
- clone oadp (we could point to a path in developer computer, but I think this way is easier to test, you can point to different fork/branch with env vars)
- run new make command `update-non-admin-manifests` (which updates the crd/rbac/samples/image of NAC of that cloned oadp) (because of this, no need for `unsupportedOverrides`)
- run `make deploy-olm` from the cloned oadp

so, if your NAC PR do change something that `update-non-admin-manifests` does not take care of (for example, you change something in NAC deployment https://github.com/openshift/oadp-operator/pull/1370/files#diff-ece703e5852171fe350087ab8bd7d5e14c7b577a83664fadc7506cafd54117e2 ) then, a simultaneously PR in oadp can/will be needed. 

Otherwise, if you just updated some NAC CRDs fields, for example, you will not have to have 2 simultaneously PRs, only one here and this new command to test NAC with OADP.

Of course, after that PR merges in NAC, a PR will be needed in OADP, which will be enforced by https://github.com/migtools/oadp-non-admin/pull/16/files#diff-c71fe913c680bf7af72b808a7695833cdbca6a10950984c003bdfc52f69555c5 and you just need to run `make update-non-admin-manifests` from compatible oadp branch to create that PR

so, in other words, in my point of view, this command removes necessity of simultaneously and blocking PRs in NAC and OADP repo and makes a little more easy/flexible to test NAC, that's why I feel it is so important

does it make sense to have the command with this explanation?

Reply:
removed command and added how to run from OADP